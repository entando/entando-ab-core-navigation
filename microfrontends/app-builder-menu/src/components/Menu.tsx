import axios from "axios";
import styled from "styled-components";
import { useEffect, useState } from "react";

const MenuCmp = styled.menu`
  position: fixed;
  left: 0;
  top: 60px;
  background-color: black;
  width: 250px;
  height: 100%;
  color: white;
`;

export function Menu() {
  const [loading, setLoading] = useState(false);
  const [message, setMessage] = useState("");

  useEffect(() => {
    const request = async () => {
      setLoading(true);

      const { data } = await axios.get<{ message: string }>("/your-endpoint");

      setMessage(data.message);
      setLoading(false);
    };

    request();
  }, []);

  return loading ? <div>{"Loading..."}</div> : <MenuCmp>{message}</MenuCmp>;
}
