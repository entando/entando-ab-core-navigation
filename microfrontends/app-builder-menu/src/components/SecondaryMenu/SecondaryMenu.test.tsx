import {
  render,
  waitForElementToBeRemoved,
  screen,
  fireEvent
} from '@testing-library/react';
import { Menu } from '../Menu';

test('Check toggling secondary menu', async () => {
  window.entando.globals = {
    ...window.entando.globals,
    userPermissions: ['managePages']
  };

  render(
    <Menu
      config={{
        systemParams: {
          api: {
            navigation: {
              url: 'http://localhost:8080'
            }
          }
        }
      }}
    />
  );

  await waitForElementToBeRemoved(screen.queryByText('Loading...'));

  expect(screen.getByText('Pages')).toBeInTheDocument();
  // Check that secondary menu is not visible
  expect(screen.queryByText('Management')).not.toBeInTheDocument();
  fireEvent.click(screen.getByText('Pages'));
  // Check that secondary menu is visible
  expect(screen.getByText('Management')).toBeInTheDocument();
  fireEvent.click(screen.getByText('Dashboard'));
  // Check that secondary menu is not visible
  expect(screen.queryByText('Management')).not.toBeInTheDocument();
  expect(window.entando.router.push).toHaveBeenCalledWith('/dashboard');
});
