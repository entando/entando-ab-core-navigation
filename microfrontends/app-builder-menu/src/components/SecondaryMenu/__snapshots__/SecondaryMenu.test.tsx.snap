// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SecondaryMenu non-visible 1`] = `
<div
  className="sc-bczRLJ lnSqPL"
>
  <div
    className="sc-gsnTZi boSiKo"
  >
    <div
      className="sc-eCYdqJ jsZDLx"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height={24}
        stroke="#39a5dc"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <polyline
          points="12 8 8 12 12 16"
        />
        <line
          x1="16"
          x2="8"
          y1="12"
          y2="12"
        />
      </svg>
    </div>
    <span
      className="sc-dkzDqf emvapn"
    >
      Secondary Menu
    </span>
  </div>
  <ul
    className="sc-hKMtZM iFxPqB"
  />
</div>
`;

exports[`renders SecondaryMenu visible 1`] = `
<div
  className="sc-bczRLJ iHxZAb"
>
  <div
    className="sc-gsnTZi boSiKo"
  >
    <div
      className="sc-eCYdqJ jsZDLx"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height={24}
        stroke="#39a5dc"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <polyline
          points="12 8 8 12 12 16"
        />
        <line
          x1="16"
          x2="8"
          y1="12"
          y2="12"
        />
      </svg>
    </div>
    <span
      className="sc-dkzDqf emvapn"
    >
      Secondary Menu
    </span>
  </div>
  <ul
    className="sc-hKMtZM iFxPqB"
  />
</div>
`;
